crates/context/primitives/src/client.rs

```diff
@@ -731,13 +734,25 @@ impl ContextClient {
         height: &NonZeroUsize,
         delta: &[u8],
     ) -> eyre::Result<()> {
+        const MAX_DELTAS_PER_MEMBER: usize = 256;
+
         let mut handle = self.datastore.handle();
 
         handle.put(
             &key::ContextDelta::new(*context_id, *public_key, height.get()),
             &types::ContextDelta::Data(delta.into()),
         )?;
 
+        // Prune old deltas to prevent unbounded storage growth
+        // Keep only the last MAX_DELTAS_PER_MEMBER deltas
+        if height.get() > MAX_DELTAS_PER_MEMBER {
+            let old_height = height.get() - MAX_DELTAS_PER_MEMBER;
+            let old_key = key::ContextDelta::new(*context_id, *public_key, old_height);
+
+            // Ignore errors on delete - the delta may have already been pruned
+            let _ignored = handle.delete(&old_key);
+        }
```

Comment:
let's instead extract it to a separate function `prune_state_delta()`?

I'd also suggest to trace errors on delete - might be useful for the future debugging.